content/blog/wallet-descriptors.md

```diff
@@ -0,0 +1,164 @@
++++
+
+Welcome back to the blog! This week we are going to be looking at the issues that led to the creation of wallet descriptors, look at their syntax and talk about their implications for wallets.
+
+## Introduction
+
+To the everyday Bitcoin user, wallets seem to be working fairly well and can easily be backed up thanks to BIP39 mnemonic phrases. Though the tree of keys wallet or the wallet generated from the mnemonic phrase often called the legacy wallet works well, it has a couple of shortcomings. Due to these shortcomings, BIP380 was proposed to resolve some of the issues faced by
+the current wallet architecture.
+
+## Why  Wallet Descriptors
+
+As mentioned in the introduction, with the old wallet we only have keys and it is not clear to the 
+wallet software which type of address types (P2PKH, P2WPKH, P2WSH, etc) to generate from these keys.
+As a consequence, wallet restoration is very computationally intensive, for each key, wallets have to scan the whole chain for different ScriptPubKey types to calculate a user's balance.
+
+Although, BIP44, BIP49 and BIP84 provide standard derivation paths to be used by all wallets, not every wallet supports these BIPs. As a direct consequence, some wallets will not be able to
+derive a user's balance given a BIP39 mnemonic phrase as they might be using a different derivation path from the wallet that produced the backup. This is will create an illusion of a loss of funds for a user thereby providing a very poor user experience.
+
+Last but not the least, this current architecture also doesn't provide the mechanism for wallets to easily watch funds in a multisig. 
+
+Last but not the least, with the advent of SEGWIT, wallets created new version bytes (version bytes are prepended to the 20 bytes of a public key to create different address types) to denote native SegWit addresses. With SEGWIT, wallets used `0x049d7cb2` for `ypub` which is an extended public key for wrapped SEGWIT addresses and `0x04b24746` for `zpub` which is an extended public key for native SegWit. Using prefixes like `xpub`, `ypub` and `zpub` are not sustainable in the long term as they are only so many letters in the alphabet.
+
+With all these problems, we need a way to help wallet software easily generate the right addresses and scripts given a particular key, extended key and derivation path. This is where wallet descriptors come to the rescue.
+
+
+## What are descriptors?
+
+Wallet descriptors often called output descriptors or output script descriptors were proposed as a new specification (BIP380) in Bitcoin which provides a plain text language that describes how outputs scripts (ScriptPubKey) and addresses are derived from keys. Wallet descriptors also contain all the information needed to spend from a particular script if the wallet software has the private key. By information, I mean the script type (P2PKH, P2WPKH, etc) and public keys necessary to create a scriptsig/witness to spend an output. 
```

Comment:
I think this (critical paragraph!) is a little unclear...
But I'll need to think in the morning about how to clarify if further 💤

Reply:
I think my issue with it is that you say "that describes how outputs scripts (ScriptPubKey) and addresses are derived from keys" but really, this just happens to be a side effect of descriptors (that any wallet could know how to generate an address from a script).

I think BIP380 summarises it best (paraphrasing the most useful chunks):

> Output Script Descriptors are a simple language which can be used to describe collections of output scripts. 
> Given just the private keys, it is not possible for restored wallets to know which kinds of output scripts and addresses to produce. This has lead to incompatibilities between wallets when restoring a backup or exporting data for a watch only wallet.
> Further complicating matters are BIP 32 derivation paths. Although BIPs 44, 49, and 84 have specified standard BIP 32 derivation paths for different output scripts and addresses, not all wallets support them nor use those derivation paths. The lack of derivation path information in these backups and exports leads to further incompatibilities between wallets.
> Current solutions to these issues have not been generic and can be viewed as being layer violations.
> Output Script Descriptors introduces a generic solution to these issues. Script types are specified explicitly through the use of Script Expressions. Key derivation paths are specified explicitly in Key Expressions. These allow for creating wallet backups and exports which specify the exact scripts, subscripts (redeemScript, witnessScript, etc.), and keys to produce.


I'd be tempted to copy this excerpt from BIP380 (with attribution) as use it in place of this paragraph, since its more correct and nicely succinct (and it'd be difficult and pointless to try to re-write in your own words).